grievance_social_protection/rights.py

```
@@ -0,0 +1,207 @@
+"""
+Rights Manager for grievance categories and flags.
+"""
+import logging
+from django.contrib.auth.models import Permission
+from django.contrib.contenttypes.models import ContentType
+from .models import Ticket
+
+logger = logging.getLogger(__name__)
+
+
+class GrievanceRightsManager:
+    """Manages automatic rights generation for grievance categories and flags"""
+
+    # Reserve 127100-127999 for dynamic grievance permissions
+    GRIEVANCE_RIGHT_BASE = 127100
+    GRIEVANCE_RIGHT_MAX = 127999
+
+    @classmethod
+    def generate_automatic_rights(cls, app_config):
+        """
+        Generate automatic rights using Django's auth_permission table.
+        Permissions are stored persistently to maintain stable IDs across config changes.
+        """
+        
+        # Get content type for Ticket model
+        try:
+            ct = ContentType.objects.get_for_model(Ticket)
+        except Exception as e:
+            logger.warning(f"Could not get ContentType for Ticket model: {e}")
+            return
```

I'd suggest leaving this Exception unhandled as if it were to happen it's something fundamentally wrong and we want to "fail fast"